packages/mui-material/src/Popper/Popper.tsx

```diff
@@ -0,0 +1,248 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import PopperUnstyled, { PopperUnstyledProps } from '@mui/core/PopperUnstyled';
+import { chainPropTypes, HTMLElementType, refType } from '@mui/utils';
+import { Direction, useThemeWithoutDefault as useTheme } from '@mui/system';
+
+export type PopperProps = PopperUnstyledProps;
+
+/**
+ *
+ * Demos:
+ *
+ * - [Autocomplete](https://mui.com/components/autocomplete/)
+ * - [Menus](https://mui.com/components/menus/)
+ * - [Popper](https://mui.com/components/popper/)
+ *
+ * API:
+ *
+ * - [Popper API](https://mui.com/api/popper/)
+ * - inherits [PopperUnstyled API](https://mui.com/api/popper-unstyled/)
+ */
+const Popper = React.forwardRef(function Popper(
+  props: PopperProps,
+  ref: React.ForwardedRef<HTMLDivElement>,
+) {
+  // destructuring only to force the propTypes generation to include all props
+  if (process.env.NODE_ENV !== 'production') {
+    const {
+      anchorEl,
+      children,
+      container,
+      direction,
+      disablePortal,
+      keepMounted,
+      modifiers,
+      open,
+      placement,
+      popperOptions,
+      popperRef,
+      style,
+      transition,
+      /* eslint-disable @typescript-eslint/no-unused-vars */
+      /* @ts-ignore */
+      ...other
+      /* eslint-enable @typescript-eslint/no-unused-vars */
+    } = props;
+  }
```

Comment:
This looks ugly but I couldn't find a way to generate propTypes without actually using the respective props. As an alternative, I could leave them only in PopperUnstyled but it could hurt the experience of using API docs.

Reply:
The reason is https://github.com/mui-org/material-ui/blob/master/docs/scripts/helpers.js#L40 When I added the code for extracting the props from the unstyled version, we didn't use `.tsx` files. this needs to be updated and then you won't need this.

Reply:
Works like a charm now, thanks a lot!